js/projects/projectsView.js

```diff
@@ -1,13 +1,35 @@
 var projectsView = {};
 
-$('#pageHeader i').on('click', function(){
+projectsView.populateProjectsFilter = function() {
+  $('article').each(function() {
+    var val = $(this).attr('data-type');
+    var optionTag = '<option value="' + val + '">' + val + '</option>';
 
-  if ($('#pageHeader nav').css('display') === 'none') {
-    $('#pageHeader nav').show();
-  } else {
-    $('#pageHeader nav').hide();
-  }
-});
+    if ($('#type-filter option[value="' + val + '"]').length === 0) {
+      $('#type-filter').append(optionTag);
+    }
+  });
+};
+
+projectsView.handleProjectsFilter = function() {
+  $('#type-filter').on('change', function(){
+    var $projects = $('article');
+    var $value = $(this).val();
+    if ($value === 'show-all') {
+      $projects.show();
+    } else if ($value) {
+      $projects.hide();
+      $projects.each(function(){
+        if ($(this).data('type') === $value) {
+          $(this).show();
+        }
+      });
+    } else {
+      $projects.show();
+    }
+    $('#type-filter').val('');
+  });
```

Comment:
This probably could be refactored clearing the select value may not be necessary anymore since you have a show all option.